Adding struct column tests

ishmandoo opened the pull request:
This PR adds some testing for combinations of nested struct and list columns.

ishmandoo commented:
Didn't see any tests along these lines, but might have missed them.

cjermain commented on py-polars/tests/test_df.py:
```diff
@@ -350,6 +350,33 @@ def test_init_records() -> None:
+def test_struct_cols() -> None:
+    """Test that struct columns can be imported and work as expected."""
+
+    def build_struct_df(data):
```
The build is failing on type checking. Do you need to add typing here? Try running `make pre-commit` in the `py-polars/` directory.

cjermain reviewed (commented):
Thanks for adding these tests. Many of the struct tests are in [`test_struct.py`](https://github.com/pola-rs/polars/blob/master/py-polars/tests/test_struct.py).

ishmandoo commented:
Moved the test to `test_struct.py`. It seems to test a super set of what was covered by `test_recursive_arrow_conversion`, so I removed that, but happy to revert it if desired.

ritchie46 commented:
Thanks @ishmandoo  and @cjermain
